sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/Transport.java

```diff
@@ -23,7 +23,7 @@
 import com.google.api.client.http.HttpRequestInitializer;
 import com.google.api.client.http.HttpTransport;
 import com.google.api.client.json.JsonFactory;
-import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.json.gson.GsonFactory;
```

Comment:
From the precommits (https://github.com/apache/beam/actions/runs/10342154630/job/28624760220?pr=32148), I see the following error:

```
Execution failed for task ':sdks:java:extensions:google-cloud-platform-core:analyzeClassesDependencies'.
> Dependency analysis found issues.
  usedUndeclaredArtifacts
   - com.google.http-client:google-http-client-gson:1.44.1@jar
  unusedDeclaredArtifacts
   - com.google.http-client:google-http-client-jackson2:1.44.1@jar
```

I think we need to replace the jackson artifact with the gson one here - https://github.com/apache/beam/blob/dd3fa04c90df95dd9998264f007f7b84bb18fe34/sdks/java/extensions/google-cloud-platform-core/build.gradle#L42

(example where we've already included the gson library - https://github.com/apache/beam/blob/780eef98083fe56f81cc5c62dc8ff193993584f0/sdks/java/io/solace/build.gradle#L50 )